### PR TITLE
Add support for group by DBRefs

### PR DESCRIPTION
**Pull request by mstrotta**
Fixes issue #584

### CODE REVIEW

File: mongomock/filtering.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     NoneType = type(None)
 
 try:
-    from bson import Regex
+    from bson import Regex, DBRef
```
Comment (pcorpet) [inline mongomock/filtering.py:19]: `mongomock` should work even if `pymongo` is not installed. Please fix what should happen in that case.

File: mongomock/filtering.py [hunk 2]
```diff
@@ -350,6 +355,8 @@ def _get_compare_type(val):
         return 45
     if isinstance(val, _RE_TYPES):
         return 50
+    if isinstance(val, DBRef):
+        return 55
```
Comment (pcorpet) [inline mongomock/filtering.py]: Thanks for adding it here as well, please make sure that the sorting of type is working as in `pymongo` by adding a line in `test__sort_mixed_types` in `test__mongomock.py`

File: tests/test__collection_api.py [hunk 3]
```diff
@@ -4273,6 +4273,26 @@ def test__aggregate_group_dict_key(self):
             list(actual)
         )
 
+    def test__aggregate_group_dbref_key(self):
```
Comment (pcorpet) [inline tests/test__collection_api.py:4329]: Thanks for adding a unit test. Please also add an equivalent test in `test__mongomock.py` to ensure consistency with `pymongo` behavior.

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks a lot for providing a PR to fix that bug. Please fix Travis and take a look at my comment.

**Comment by kosqx** [comment]
Hi! Your CI build failed because of the https://github.com/mongomock/mongomock/issues/648 error.

I will push the fix soon

**Comment by kosqx** [comment]
The fix is ready, tests are passing: https://github.com/mongomock/mongomock/pull/649

**Pull request merged**